From 6db43a1417f73e588ea1935535874c9875e1d459 Mon Sep 17 00:00:00 2001
From: Min He <min.he@intel.com>
Date: Sat, 18 Mar 2017 03:31:13 +0800
Subject: [PATCH 0909/2313] drm/i915/gvt: Initiazlie BXT pipe info early for
 GVT-g

GVT will utilize the num_sprite to initialize the mmio info, but
num_sprite is initialized later than GVT initialization, so move
the pipe info earlier.

This patch can fix some uninitialized mmio in different planes.

Signed-off-by: Min He <min.he@intel.com>
Reviewed-by: Singh, Satyeshwar <satyeshwar.singh@intel.com>
(cherry picked from commit 155c8b5f8df597b52649f689e5a187d0e30eb889)

Change-Id: Ib9cfd60750c4c7e26788f2389d779bf165a8621f
Reviewed-by: He, Min <min.he@intel.com>
Reviewed-by: Jiang, Fei <fei.jiang@intel.com>
Reviewed-by: Dong, Eddie <eddie.dong@intel.com>
Tested-by: Dong, Eddie <eddie.dong@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index c75f4cc..133ce09 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -889,6 +889,14 @@ static int i915_driver_init_early(struct drm_i915_private *dev_priv,
 	if (ret < 0)
 		goto err_engines;
 
+	if (IS_BROXTON(dev_priv)) {
+		struct intel_device_info *info = mkwrite_device_info(dev_priv);
+
+		info->num_sprites[PIPE_A] = 2;
+		info->num_sprites[PIPE_B] = 2;
+		info->num_sprites[PIPE_C] = 1;
+	}
+
 	/* This must be called before any calls to HAS_PCH_* */
 	intel_detect_pch(dev_priv);
 
-- 
2.7.4

